File: src/main/java/me/wawwior/utils/hashing/Hash.java
package me.wawwior.utils.hashing;

import java.nio.charset.Charset;

public interface Hash {

    default byte[] hash(String string) {
        return hash(string, Charset.defaultCharset());
    }

    default byte[] hash(String string, Charset charset) {
        return hash(string.getBytes(charset));
    }

    byte[] hash(byte[] bytes);

    boolean compare(byte[] bytes, byte[] hash);

    int hashLength();

}

File: src/main/kotlin/me/wawwior/utils/hashing/Hash.kt
package me.wawwior.utils.hashing

import java.nio.charset.Charset

interface Hash {

    fun hash(string: String): ByteArray {
        return hash(string, Charsets.UTF_8)
    }

    fun hash(string: String, charset: Charset): ByteArray {
        return hash(string.toByteArray(charset))
    }

    fun hash(bytes: ByteArray): ByteArray

    fun compare(bytes: ByteArray, hash: ByteArray): Boolean

    fun hashLength(): Int

}